packages/vue3/src/components/Upload/FileInput.vue

```diff
@@ -1,94 +1,65 @@
 <script setup lang="ts">
-import { ref, useSlots } from 'vue';
+import { computed, useSlots } from 'vue';
 import BaseFileInput from './Internal/BaseFileInput.vue';
 
-withDefaults(
-  defineProps<{ size?: 'sm' | 'md' | 'lg'; limit?: number }>(),
-  { size: 'md', limit: 1 },
+const props = withDefaults(
+  defineProps<{
+    modelValue: File[]
+    limit?: number
+    disabled?: boolean
+  }>(),
+  {
+    limit: 1,
+    disabled: false,
+  },
 );
 
-const emit = defineEmits(['input']);
-const slots = useSlots();
+const emit = defineEmits(['update:modelValue']);
 
-const files = ref<File[]>([]);
-const dragging = ref<boolean>(false);
+const model = computed({
+  get: () => props.modelValue,
+  set: (value: File[]) => {
+    emit('update:modelValue', value);
+  },
+});
 
-function handlefiles(files: File[]) {
-  dragging.value = false;
-  emit('input', files);
-}
+const slots = useSlots();
 </script>
 
 <template>
   <BaseFileInput
```

Comment:
Can you fix the indentation here so that it's all in one line
```suggestion
  <BaseFileInput v-model="model" :limit="limit" :disabled="disabled">
```